analyzers/tests/SonarAnalyzer.UnitTest/TestCases/TestMethodShouldContainAssertion.NUnit.cs

```diff
@@ -519,6 +521,53 @@ public void ReceivedInOrder()
         }
     }
 
+    // All assert methods start with "Should*". We do not test all APIs, just some.
+    public class ShouldlyTests
+    {
+        int i = 1;
+        public dynamic Foo() => null;
+        public void Bar() { }
+
+        [TestCase]
+        public void NoAssert() // Noncompliant
+        {
+        }
+
+        [TestCase]
+        public void W()
+        {
+            int[] empty = { };
+            empty.ShouldBeEmpty();
+
+            var dict = new Dictionary<string, string> { { "x", "x" } };
+            dict.ShouldContainKey("y");
+
+            Should.Throw<IndexOutOfRangeException>(() => Bar());
+        }
+
+        [TestCase]
+        public void X() => "Foo".ShouldMatchApproved();
+
+        [TestCase]
+        public void Y() => i.ShouldBeInRange(30000, 40000);
+
+        [TestCase]
+        public void Z() => Should.NotThrow(() => Bar());
+
+        [TestCase]
+        public void Dynamic()
+        {
+            dynamic theFuture = Foo();
+            DynamicShould.HaveProperty(theFuture, "X");
```

Comment:
this method doesn't start with "should" but the logic actually splits the full expression in words and verifies if any is "should"